src/java.net.http/share/classes/jdk/internal/net/http/Stream.java

```diff
@@ -187,7 +190,13 @@ private void schedule() {
                 Http2Frame frame = inputQ.peek();
                 if (frame instanceof ResetFrame rf) {
                     inputQ.remove();
-                    handleReset(rf, subscriber);
+                    if (endStreamReceived()) {
```

Comment:
```suggestion
                    if (endStreamReceived() && rf.getErrorCode() == ResetFrame.NO_ERROR) {
```

Reply:
So if we have received an end stream but receive an error code, do we want to complete exceptionally? 

The suggestion seems correct, just checking 👍

Reply:
Yes - I believe so - the peer indicates an error - so we should complete exceptionally